lib/ansible/modules/cloud/azure/azure_rm_policysetdefinition_facts.py

```diff
@@ -0,0 +1,239 @@
+            - To gather all built-in policy set definitions when C(built_in) is set to True
+            - This parameter is mutually exclusive with 'management_group'.
+        type: bool
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Fan Qiu (@MyronFanQiu)"
+
+'''
+
+EXAMPLES = '''
+- name: List all policy set definitions
+  azure_rm_policysetdefinition_facts:
+
+- name: List all built-in policy set definitions
+  azure_rm_policysetdefinition_facts:
+    built_in: true
+
+- name: List all built-in policy set definitions
+  azure_rm_policysetdefinition_facts:
+    name: testpolicysetdefinition
+    built_in: true
+'''
+
+RETURN = '''
+policysetdefinitions:
+    description: List of policy set definitions.
+    returned: always
+    type: list
+    contains:
+        description:
+            description:
+                - The policy set definition description.
+            type: str
+            sample: "This policy audits Linux virtual machines that do not have the specified applications installed."
+        display_name:
+            description:
+                - The display name of the policy set definition.
+            type: str
+            sample: "Audit Linux VMs that do not have the specified applications installed"
+        id:
+            description:
+                - The ID of the policy set definition.
+            type: str
+            sample: "/providers/Microsoft.Authorization/policySetDefinitions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+        metadata:
+            description:
+                - The policy set definition metadata.
+            type: dict
+            sample: '{
+                "category": "Guest Configuration"
+            }'
+        name:
+            description:
+                - The name of the policy set definition.
+            type: str
+            sample: "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+        parameters:
+            description:
+                - Required if a parameter is used in policy set rule.
+            type: dict
+            sample: '{
+                "effect": {
+                    "allowedValues": [
+                        "AuditIfNotExists",
+                        "Disabled"
+                    ],
+                    "defaultValue": "AuditIfNotExists",
+                    "metadata": {
+                        "description": "Enable or disable the execution of the pol
+                        "displayName": "Effect"
+                    },
+                    "type": "String"
+                }
+            }'
+        policy_type:
+            description:
+                - The type of policy definition.
+                - Possible values are NotSpecified, BuiltIn, and Custom.
+            type: str
+            sample: "BuiltIn"
+        type:
+            description:
+                - Resource type
+            sample: "Microsoft.Authorization/policySetDefinitions"
+            type: str
+        policy_definitions:
```

Comment:
what's the difference of the id in policy_defintions with `id` field?

Reply:
Just as the name. The policy setdefinition is a set of policy definitions. So the id in the policy_definitions is the id of the policy definition. The ```id``` field is the id of the policy setdefinition.